frontend/src/metabase/ui/components/inputs/TimeInput/TimeInput.tsx

```diff
@@ -0,0 +1,61 @@
+import { useLayoutEffect, useState } from "react";
+import type { ChangeEvent } from "react";
+import moment from "moment-timezone";
+import { TimeInput as MantineTimeInput } from "@mantine/dates";
+import type { TimeInputProps as MantineTimeInputProps } from "@mantine/dates";
+
+const TIME_FORMAT = "HH:mm";
+
+export type TimeInputProps = Omit<
+  MantineTimeInputProps,
+  "value" | "defaultValue" | "onChange"
+> & {
+  value?: Date | null;
```

Comment:
`null` is allowed to be compatible with `DateInput`. It has `clearable` option (`false` by default).